sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -76,7 +76,7 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
       case (TIMESTAMP_LTZ, Nil) => TimestampType
       case (STRING, Nil) =>
         typeCtx.children.asScala.toSeq match {
-          case Seq(_) => SqlApiConf.get.defaultStringType
+          case Seq(_) => DefaultStringType()
```

Comment:
I'm really worried about this. We used to return `StringType` object if collation is not enabled. But now it becomes `DefaultStringType`. This is kind of a breaking change to the parser.

Reply:
This doesn't only affect DDL commands but everything that needs a data type, e.g. CAST

Reply:
I think we should only use this `DefaultStringType` in the targeting DDL commands.

Reply:
Is it a breaking change even though the `DefaultStringType() == StringType` will still yield true? It's really just a wrapper with exactly the same functionality and semantics

Reply:
Although I see your point because we can't do:
```
DefaultStringType() match {
  case StringType =>
    ...
} 
```

Reply:
@cloud-fan is [this](https://github.com/apache/spark/pull/48436/files#diff-26fe6b511f4b8aedf5bdffae0d7c03aba69463c09fa4f881d77520125796e0faR3318) then problematic as well?

Reply:
Whatever annotation we put into `StringType` (a subclass, a new flag, etc.), it can be a breaking change. I think it's safer to only add the annotation when needed (DDL commands).